### PR TITLE
Core-eval governance support

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

New features and tests to help make core-eval governance more reproducible.

### CODE REVIEW

File: packages/deploy-script-support/src/getBundlerMaker.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      log('Instantiating bundle maker...');
+      const { publicFacet } = await E(zoe).startInstance(
+        E(zoe).install(bundle),
+      );
+      bundlerMaker = publicFacet;
+      await E(scratch).set('bundlerMaker', bundlerMaker);
+    }
+
+    if (board) {
+      const bundlerMakerId = await E(board).getId(bundlerMaker);
```
Comment (dckc) [inline packages/deploy-script-support/src/getBundlerMaker.js]: this `await` is not top level. worth a comment to say that this is not production code? either here or at the top of the file or both?

File: packages/deploy-script-support/src/getBundlerMaker.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// @ts-check
+import { E } from '@endo/far';
+import url from 'url';
+
+export const makeGetBundlerMaker =
+  ({ board, zoe, scratch }, { lookup, bundleSource }) =>
```
Comment (dckc) [inline packages/deploy-script-support/src/getBundlerMaker.js]: lack of types obscures the fact that board is optional. hm.

File: packages/deploy-script-support/src/getBundlerMaker.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// @ts-check
```
Comment (dckc) [inline packages/deploy-script-support/src/getBundlerMaker.js:1]: Add a note that this should go away after we have bundlecaps, with a pointer to and from a relevant issue?

**Review by dckc — APPROVED** [review]
presuming you have tested that this is useful in your oracle work, looks good.

my comments are not critical

**Review by JimLarson — APPROVED** [review]
The Go file LGTM.

File: packages/cosmic-swingset/scripts/clean-core-eval.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  const clean = defangEvaluableCode(text);
+  const withURL = `${clean}\n//# sourceURL=${fn}\n`;
+
+  // end-of-line whitespace disrupts YAML formatting
+  const trimmed = withURL.replace(/[\r\t ]+$/gm, '');
+  compartmentEvaluate(trimmed);
+  stdout.write(trimmed);
+};
+
+if (isEntrypoint(import.meta.url)) {
```
Comment (dckc) [inline packages/cosmic-swingset/scripts/clean-core-eval.js:62]: interesting approach. `isEntrypoint` is powerful, but I suppose so was `require.main`.

A [Stackoverflow post](https://stackoverflow.com/a/68848622/7963) has a succinct version:

```
if (import.meta.url === `file://${process.argv[1]}`) {
  // module was not imported but called directly
}
```

but that's less robust; it's followed by one that uses `url.pathToFileURL` as in `isEntrypoint`.

**Pull request merged**